edit_book.php

```diff
@@ -0,0 +1,44 @@
+<?php
+include 'database.php';
```

Comment:
**Observations**: The SQL queries are vulnerable to SQL injection due to improper handling of input parameters.

**Recommendations**: Use prepared statements to secure the queries. Validate the `id` parameter to ensure it's an integer.

**Example**:
```php
if (isset($_GET['id'])) {
    $id = (int)$_GET['id'];
    $query = "SELECT * FROM books WHERE id = ?";
    $stmt = mysqli_prepare($connection, $query);
    mysqli_stmt_bind_param($stmt, 'i', $id);
    mysqli_stmt_execute($stmt);
    $result = mysqli_stmt_get_result($stmt);
    $book = mysqli_fetch_assoc($result);
}

if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $id = (int)$_POST['id'];
    $title = mysqli_real_escape_string($connection, $_POST['title']);
    $author = mysqli_real_escape_string($connection, $_POST['author']);
    $year = (int)$_POST['year'];

    $query = "UPDATE books SET title = ?, author = ?, year = ? WHERE id = ?";
    $stmt = mysqli_prepare($connection, $query);
    mysqli_stmt_bind_param($stmt, 'ssii', $title, $author, $year, $id);
    if (mysqli_stmt_execute($stmt)) {
        echo "Book updated successfully!";
        header("Location: list_books.php");
    } else {
        echo "Error: " . mysqli_stmt_error($stmt);
    }
}
```